src/NativeScript/Calling/FFICallPrototype.cpp

```diff
@@ -0,0 +1,51 @@
+//
+//  FFICallPrototype.cpp
+//  NativeScript
+//
+//  Created by Yavor Georgiev on 25.09.15.
+//  Copyright (c) 2015 Telerik. All rights reserved.
+//
+
+#include "FFICallPrototype.h"
+#include "FFICall.h"
+
+namespace NativeScript {
+using namespace JSC;
+
+const ClassInfo FFICallPrototype::s_info = { "FFIFunction", &Base::s_info, 0, CREATE_METHOD_TABLE(FFICallPrototype) };
+
+EncodedJSValue JSC_HOST_CALL FFICallPrototypeFuncAsync(ExecState*);
+
+void FFICallPrototype::finishCreation(VM& vm, JSGlobalObject* globalObject) {
+    Base::finishCreation(vm);
+    vm.prototypeMap.addPrototype(this);
+
+    JSC_NATIVE_FUNCTION(Identifier::fromString(&vm, "async"), FFICallPrototypeFuncAsync, DontEnum, 0);
+}
+
+    EncodedJSValue JSC_HOST_CALL FFICallPrototypeFuncAsync(ExecState* execState) {
+        auto call = jsCast<FFICall*>(execState->thisValue());
+
+        JSValue array = execState->argument(1);
+
+        MarkedArgumentBuffer applyArgs;
+        if (!array.isUndefinedOrNull()) {
+            if (!array.isObject())
+                return throwVMTypeError(execState);
+            if (isJSArray(array)) {
+                if (asArray(array)->length() > maxArguments)
```

Comment:
I literally copied this from the old implementation of [`JSC::FunctionPrototype::functionProtoFuncApply`](https://github.com/WebKit/webkit/blob/fb7eff208db0fa9af1a929c6b6e26a32310caca3/Source/JavaScriptCore/runtime/FunctionPrototype.cpp#L127). `maxArguments` is a const declared in `Interpreter.h`.